File: app/src/main/java/com/john/kot/arch/viewmodel/UserModel.java
package com.john.kot.arch.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.LinkedList;
import java.util.List;

public class UserModel extends ViewModel {

    private final MutableLiveData userLiveData = new MutableLiveData();
    List<User> userList = new LinkedList<>();

    public LiveData getUser() {
        return userLiveData;
    }

    public UserModel() {
    }

    void doAction() {
        userList.add(new User("lili"));
        userLiveData.postValue(userList);
    }
}

File: app/src/main/java/com/kot/mvvm/viewmodel/UserModel.kt
package com.kot.mvvm.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import java.util.LinkedList

class UserModel : ViewModel() {
    private val _userLiveData = MutableLiveData<List<UserObject>>()
    val userLiveData: LiveData<List<UserObject>> = _userLiveData

    private val userList = LinkedList<UserObject>()

    fun doAction() {
        userList.add(UserObject("lili"))
        _userLiveData.postValue(userList)
    }
}